ext/standard/tests/mail/gh13415.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+GH-13415 (Added validation of line breaks \n in $additional_headers of mail())
+--INI--
+sendmail_path={MAIL:gh13415.out}
+--FILE--
+<?php
+mail('to@example.com', 'Test Subject', 'A Message', ['Reply-To' => "foo@example.com \nCc: hacker@example.com"]);
+?>
+--CLEAN--
+<?php
+if (file_exists('gh13415.out')) {
+    unlink('gh13415.out');
+}
+?>
+--EXPECTF--
+Fatal error: Uncaught ValueError: Header "Reply-To" has invalid format, or contains invalid characters in %s
```

Comment:
we should consider some message improvement to indicate what's wrong when we add more cases that are valid in terms of RFC 2822.

Reply:
Agree. We probably need to change the message in `php_mail_build_headers_elem`.

It may be necessary to change `php_mail_build_headers_check_field_name` and `php_mail_build_headers_check_field_value` to return Enums instead of bools so that the type of error can be determined.

Reply:
So it could be something like following:
- LF only: `Header "Reply-To" contains new line character that is as used a separator supported by most message transport agents`
- CR only: `Header "Reply-To" contains carriage return character that is used as a separator supported by most message transport agents` (I'm not actually sure if it's used as a separator so if it's not it could be just `...that is not handled by most ...`)
- CRLF: `Header "Reply-To" contains carriage return character followed by new line character that must be used as a line separator and is not allowed in the header`
- \0 in the message: `Header "Reply-To" contains null character that is not allowed in the header`

The wording is not great and we might consider replacing new line with NL (or LF) and  carriage return with CR but it is better than the current message.

Reply:
Actually just checking https://www.rfc-editor.org/rfc/rfc5322#section-2.2 which explicitly prohibits CR and LF in header field so it's not just because some MTA's. So it should all contain it should be more:
LF only: `Header "Reply-To" contains LF character that is not allowed in the header`
CR only: `Header "Reply-To" contains CR character that is not allowed in the header`
CRLF: `Header "Reply-To" contains CRLF characters that are used as a line separator and are not allowed in the header`